File: backend/models/record_model.py
from bson import ObjectId
from datetime import datetime

# ----------------------------
# SINGLE RECORD MODEL
# ----------------------------
def record_model(record) -> dict:
    if not record:
        return {}

    return {
        "id": str(record.get("_id")) if record.get("_id") else None,

        "doctor": record.get("doctor", ""),
        "patient": record.get("patient", ""),

        "diagnosis": record.get("diagnosis", ""),
        "prescription": record.get("prescription", ""),
        "notes": record.get("notes", ""),

        "created_at": format_date(record.get("created_at")),
        "updated_at": format_date(record.get("updated_at")),
    }


# ----------------------------
# MULTIPLE RECORDS MODEL
# ----------------------------
def records_model(records) -> list:
    return [record_model(record) for record in records]


# ----------------------------
# DATE FORMAT HELPER
# ----------------------------
def format_date(date):
    if isinstance(date, datetime):
        return date.isoformat()
    return None

File: lib/models/record_model.dart
class RecordModel {
  final String? id;
  final String patient;
  final String? doctor;
  final String diagnosis;
  final String prescription;
  final DateTime? createdAt;

  RecordModel({
    this.id,
    required this.patient,
    this.doctor,
    required this.diagnosis,
    required this.prescription,
    this.createdAt,
  });

  // ----------------------------
  // FROM JSON (BACKEND → FLUTTER)
  // ----------------------------
  factory RecordModel.fromJson(Map<String, dynamic> json) {
    return RecordModel(
      id: json["_id"]?.toString() ?? json["id"]?.toString(),
      patient: json["patient"] ?? "",
      doctor: json["doctor"],
      diagnosis: json["diagnosis"] ?? "",
      prescription: json["prescription"] ?? "",
      createdAt: json["created_at"] != null
          ? DateTime.tryParse(json["created_at"].toString())
          : null,
    );
  }

  // ----------------------------
  // TO JSON (FLUTTER → BACKEND)
  // ----------------------------
  Map<String, dynamic> toJson() {
    return {
      "patient": patient,
      "diagnosis": diagnosis,
      "prescription": prescription,
    };
  }
}